### PR TITLE
Property details

### PR DESCRIPTION
**Pull request by Hammond1979**
- Feedback worked on

### CODE REVIEW

**Comment by nosisky** [comment]
We have reviewed, the authentication routes before now but I can still see the changes in this PR.

File: src/controllers/agent.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    ` where email = '${email}' `
+  );
+    if (checkEmail.rowCount > 0 ) {
+      return res.status(409).json({ message: 'Account already Exist' });
+    }
+    const data = await agentModel.insertWithReturn(columns, values);
+    const {id} = data.rows[0]
+    const newUser = { id, firstName, lastName, email};
+
+  const token = jwt.sign({ newUser}, process.env.TOKEN_KEY, {
```
Comment (nosisky) [inline src/controllers/agent.js]: This logic to sign JWT should be in a helper function.

File: src/controllers/agent.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+export const agentLogin = async (req, res) => {
+  const { email, password } = req.body;
+  try {
+    const validEmail = await agentModel.select('*', ` WHERE  email = '${email}' `);
+    if (!validEmail.rows.length) return res.status(400).json({ messages: 'Email or password supplied is invalid' });
+    const validPassword = await bcrypt.compare(password, validEmail.rows[0].password);
+    if (!validPassword) return res.status(400).json({ message: 'Email or password supplied is invalid' });
+    const user = { email };
```
Comment (nosisky) [inline src/controllers/agent.js]: Why are you signing just the user email? how do you hope to get the userId later?

File: src/controllers/agent.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    const token = jwt.sign({ user}, process.env.TOKEN_KEY, {
+      expiresIn: '30m',
```
Comment (nosisky) [inline src/controllers/agent.js]: Please put the logic to sign the JWT in a helper function so you dont have to repeat yourself.

File: src/controllers/agent.js (hunk 1 above)
Comment (Hammond1979) [reply src/controllers/agent.js]: Resolved already moved too helper function.

File: src/controllers/agent.js (hunk 3 above)
Comment (Hammond1979) [reply src/controllers/agent.js]: Resolved also...

File: src/controllers/agent.js (hunk 2 above)
Comment (Hammond1979) [reply src/controllers/agent.js]: i have more features too it so the login wouldn't appear as only the user email.

File: src/controllers/propertyDetails/property.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  } catch (err) {
+    res.status(500).json({ messages: err.stack.messages });
+  }
+};
+
+export const allProperties = async (req, res) => {
+  try {
+    const getProperties = await propertyModel.select('*');
+    if (getProperties.rows === 0) {
+      return res.status(409).json({ message: 'no property' });
```
Comment (nosisky) [inline src/controllers/propertyDetails/property.js]: 409 is a wrong status code for when data is not found, the right thing should be 404.

Comment (nosisky) [reply src/controllers/propertyDetails/property.js]: Please read up more on HTTP status code.

File: src/controllers/propertyDetails/property.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  } catch (err) {
+    res.status(500).json({ messages: err.stack.messages });
+  }
+};
+export const getPropertyById = async (req, res) => {
+  const { id } = req.params;
+  try {
+    const getProperty = await propertyModel.select('*', ` WHERE  id = '${id}' `);
+    if (getProperty.rows === 0) {
+      return res.status(409).json({ message: 'no property' });
```
Comment (nosisky) [inline src/controllers/propertyDetails/property.js]: Same status code issue here.

File: src/controllers/propertyDetails/property.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  }
+  next();
+};
+export const deleteProperty = async (req, res) => {
+  const { id } = req.params;
+  const agentId = req.user.user.id;
+  try {
+    const getProperty = await propertyModel.delete(` WHERE  id = '${id}' `);
+    if (getProperty.rows === 0) {
+      return res.status(409).json({ message: 'no property' });
```
Comment (nosisky) [inline src/controllers/propertyDetails/property.js]: Same status code issue here.

Comment (nosisky) [reply src/controllers/propertyDetails/property.js]: Why are you also getting property that has already been deleted?

File: src/controllers/propertyDetails/property.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  } catch (err) {
+    res.status(500).json({ messages: err.stack });
+  }
+  next();
+};
+export const deleteProperty = async (req, res) => {
+  const { id } = req.params;
+  const agentId = req.user.user.id;
+  try {
+    const getProperty = await propertyModel.delete(` WHERE  id = '${id}' `);
```
Comment (nosisky) [inline src/controllers/propertyDetails/property.js]: We can't name this variable `getProperty` when the action assigned to it is a delete property. There is really no need to assign a delete operation to a variable.

File: src/controllers/propertyDetails/property.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    if (getProperty.rows === 0) {
+      return res.status(409).json({ message: 'no property' });
+    }
+    res.status(200).json({message: 'property is deleted successfully'});
+  } catch (err) {
+    res.status(500).json({ messages: err.stack.messages });
+  }
+};
+export const agentProperties = async (req, res) => {
```
Comment (nosisky) [inline src/controllers/propertyDetails/property.js]: This is a controller, it should be named something like `getAgentProperties`

Comment (Hammond1979) [reply src/controllers/propertyDetails/property.js]: feedback worked on

**Pull request merged**